### PR TITLE
Improve code quality with static analysis

### PR DESCRIPTION
**Pull request by EwoutH**
I ran some static analysis over the codebase (using https://deepsource.io/) to improve the code quality. It mainly removes some unnecessary complex constructs, like lambda expressions, list/dict comprehension and super class methods, where they aren't necessary. This improves performance and readability.

Each commit has a description what is changed and why. If there are certain things anyone doesn't agree on, please let it know!

### CODE REVIEW

File: examples/color_patches/color_patches/server.py [hunk 1]
```diff
@@ -46,7 +46,8 @@ def color_patch_draw(cell):
     :return: the portrayal dictionary.
 
     """
-    assert cell is not None
+    if cell is None:
```
[the remark this answers is not in the document]
Comment (rht) [reply examples/color_patches/color_patches/server.py:49]: This gives more context: https://www.reddit.com/r/Python/comments/9vj59a/why_you_should_never_use_assert_in_python/.
To re-emphasize, running Python with `-O` or `-OO` will strip those assertion lines.

**Comment by rht** [comment]
The LOC changes are pretty small (~20), so this means the Mesa codebase is clean ;)

**Comment by rht** [comment]
Why isn't deepsource.io analyzing the JS files?

**Comment by EwoutH** [comment]
Thanks for the quick review and merging. Indeed, the Mesa codebase is quite clean!

> Why isn't deepsource.io analyzing the JS files?

I had it only configured for Python files, would you like also running it on JavaScript files?

**Comment by rht** [comment]
Yes, analyzing the JS files would be great. But I'd predict to have lots of gotchas, because the JS files are old (written during the jQuery era), and haven't been touched for a while.

**Comment by rht** [comment]
I will first clean the JS files, e.g. remove jQuery usage, use ES6 module import instead of `<script>` tag, etc. Let's try again in a month from now.

**Comment by rht** [comment]
Does it require a sign up for the `projectmesa` org? I tried signing up for deepsource.io, but it requiring the permission to commit on my behalf is rather scary.

**Comment by EwoutH** [comment]
Yeah that is true. It creates branches for each feature. I only have it enabled on a select number of forks thus.

I can also just run it weekly.

**Comment by rht** [comment]
Weekly for the Python code (for now, but including JS later) sounds good to me.

**Pull request merged**